File: java/0045.jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int end = 0;
        int max_P = 0;
        int steps = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            max_P = Math.max(max_P, i + nums[i]);
            if (i == end) {
                end = max_P;
                steps++;
            }
        }
        return steps;
    }
}

File: python/0045. Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        maxPos,end,step = 0,0,0
        for i in range(n-1):
            maxPos = max(maxPos, i + nums[i])
            if i == end:
                end = maxPos
                step+=1
        return step
